test/Microsoft.AspNetCore.OData.E2E.Tests/Lists/ListsController.cs

```diff
@@ -0,0 +1,179 @@
+        }
+
+        [EnableQuery(PageSize = 10, MaxExpansionDepth = 5)]
+        public IActionResult Get()
+        {
+            return Ok(_dbContext.Set<Product>());
+        }
+
+        public IActionResult Get(string key)
+        {
+            return Ok(_dbContext.Set<Product>().Find(key));
+        }
+
+        public IActionResult Post([FromBody] Product Product)
+        {
+            Product.ProductId = _dbContext.Set<Product>().Count() + 1+"";
+            _dbContext.Set<Product>().Add(Product);
+            _dbContext.SaveChanges();
+
+            return Created(Product);
+        }
+
+        public IActionResult Put(int key, [FromBody] Product Product)
+        {
+            Product.ProductId = key+"";
+            Product originalProduct = _dbContext.Set<Product>().Find(key);
+
+            if (originalProduct == null)
+            {
+                _dbContext.Set<Product>().Add(Product);
+
+                return Created(Product);
+            }
+
+            _dbContext.Set<Product>().Remove(originalProduct);
+            _dbContext.Set<Product>().Add(Product);
+            return Ok(Product);
+        }
+
+        public IActionResult Patch(int key, [FromBody] Delta<Product> delta)
+        {
+            Product originalProduct = _dbContext.Set<Product>().Find(key);
+
+            if (originalProduct == null)
+            {
+                Product temp = new Product();
+                delta.Patch(temp);
+                _dbContext.Set<Product>().Add(temp);
+                return Created(temp);
+            }
+
+            delta.Patch(originalProduct);
+            return Ok(delta);
+        }
+
+        public IActionResult Delete(int key)
+        {
+            Product Product = _dbContext.Set<Product>().Find(key);
+
+            _dbContext.Set<Product>().Remove(Product);
+            return this.StatusCode(StatusCodes.Status204NoContent);
+        }
+
+        [HttpPost("ResetDataSource")]
+        public IActionResult ResetDataSource()
+        {
+            _dbContext.Set<Product>().RemoveRange(_dbContext.Set<Product>());
+            _dbContext.SaveChanges();
+
+            // Add new seed data
+            _dbContext.Set<Product>().AddRange(
+               new Product
+               {
+                   ProductId = "1",
+                   Name = "Product1",
+                   Category = "Category1",
+                   ListTestString = new List<string> { "Test1", "Test2", "Test99" },
+                   ListTestBool = new List<bool> { true, false },
+                   ListTestInt = new List<int> { 1, 2, 3 },
+                   ListTestDouble = new List<double> { 1.1, 2.2 },
+                   ListTestFloat = new List<float> { 1.1f, 2.2f },
+                   ListTestDateTime = new List<DateTimeOffset> { DateTime.Now, DateTime.UtcNow },
+                   ListTestUri = new List<Uri> { new Uri("https://example.com") },
+                   ListTestUint = new uint[] { 1, 2, 3 }
```

Comment:
Initialize `ListTestOrders` with data here, and leave `ListTestOrders` as null for the other products 
```suggestion
                   ListTestUint = new uint[] { 1, 2, 3 },
                   ListTestOrders = new List<Order>
                    {
                        new Order { OrderId = "Order1" },
                        new Order { OrderId = "Order2" }
                    }
```